ios/FluentUI/Notification/MSFNotificationView.swift

```diff
@@ -0,0 +1,182 @@
+        if let anchorView = anchorView, anchorView.superview == view {
+            view.insertSubview(hostingControllerView, belowSubview: anchorView)
+        } else {
+            view.addSubview(hostingControllerView)
+        }
+
+        let anchor = anchorView?.topAnchor ?? view.safeAreaLayoutGuide.bottomAnchor
+        constraintWhenHidden = hostingControllerView.topAnchor.constraint(equalTo: anchor)
+        constraintWhenShown = hostingControllerView.bottomAnchor.constraint(equalTo: anchor, constant: -presentationOffset)
+
+        var constraints = [NSLayoutConstraint]()
+        constraints.append(animated ? constraintWhenHidden : constraintWhenShown)
+        if style.needsFullWidth {
+            constraints.append(hostingControllerView.leadingAnchor.constraint(equalTo: view.leadingAnchor))
+            constraints.append(hostingControllerView.trailingAnchor.constraint(equalTo: view.trailingAnchor))
+        } else {
+            constraints.append(hostingControllerView.centerXAnchor.constraint(equalTo: view.centerXAnchor))
+            constraints.append(hostingControllerView.widthAnchor.constraint(lessThanOrEqualTo: view.safeAreaLayoutGuide.widthAnchor, constant: -2 * presentationOffset))
+        }
+        NSLayoutConstraint.activate(constraints)
+
+        if style.isToast {
+            MSFNotification.currentToast = self
+        }
+
+        let completionForShow = { (_: Bool) in
+            UIAccessibility.post(notification: .layoutChanged, argument: hostingControllerView)
+            completion?(self)
+        }
+
+        if animated {
+        view.layoutIfNeeded()
+            UIView.animate(withDuration: style.animationDurationForShow, delay: 0, usingSpringWithDamping: style.animationDampingRatio, initialSpringVelocity: 0, animations: {
+                self.constraintWhenHidden.isActive = false
+                self.constraintWhenShown.isActive = true
+                view.layoutIfNeeded()
+            }, completion: completionForShow)
+        } else {
+            completionForShow(true)
+        }
+    }
+
+    @objc public func hide(after delay: TimeInterval = 0, completion: (() -> Void)? = nil) {
+        var updatedDelay = delay
+        if updatedDelay != 0 {
+            switch self.state.style {
+            case .primaryToast, .primaryBar, .primaryOutlineBar, .neutralBar:
+                break
+            default:
+                updatedDelay = .infinity
+            }
+        }
+
+        guard self.view.window != nil && updatedDelay != .infinity else {
+            return
+        }
+
+        if updatedDelay > 0 {
+            DispatchQueue.main.asyncAfter(deadline: .now() + updatedDelay) { [weak self] in
+                self?.hide(completion: completion)
+            }
+            return
+        }
+
+        let hostingControllerView = self.hostingController.view
+        if let completion = completion {
+            completionsForHide.append(completion)
```

Comment:
Why is completionsForHide an array? Is it calling completions for other toasts that were previously requested to be hidden?

Reply:
I'm not sure, it was how whoever created the original UIKit version of Notification had implemented it, so I left it.